CRM/Price/BAO/PriceSet.php

```diff
@@ -1633,14 +1633,14 @@ protected static function addPriceFieldsToForm(CRM_Core_Form $form, $feeBlock, b
         !$validFieldsOnly
       ) {
         $options = $field['options'] ?? NULL;
-        if ($className == 'CRM_Contribute_Form_Contribution_Main' && $component = 'membership') {
+        if ($className == 'CRM_Contribute_Form_Contribution_Main' && $component == 'membership') {
```

Comment:
Fixing this was not necessary, and maybe should be in a separate PR. Looks like a bug?